core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -83,7 +83,7 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
         case _: AggregateWorkspaceNotFoundException =>
           // if a WSM workspace does not already exist, assume the platform is GCP, confirm platform compatibility,
           // and then create a stub workspace
-          snapshotValidator.validateWorkspacePlatformCompatibility(WorkspaceCloudPlatform.Gcp)
+          snapshotValidator.validateWorkspacePlatformCompatibility(Some(WorkspaceCloudPlatform.Gcp))
```

Comment:
1) If I understand your comments correctly, it sounds like `wsmWorkspace` will _always_ have `getCloudPlatform` if the workspace state is `READY`; otherwise `fetchAggregatedWorkspace` will throw an error.  Do I have that right?

2) I also think maybe there's a business logic rule that isn't codified here explicitly (correct me if I'm wrong though as I'm not sure if my hunch is correct), which is that we should only create snapshots for READY workspaces.

If both of my assertions are correct, then I wonder if you can avoid having to make any changes within `validateWorkspacePlatformCompatibility` by just having this try block check the WSM state is READY and fail for that reason (rather than later on due to the omission of the platform).

As long as both (1) and (2) are actually true, I think this might be a bit more intention revealing.

Reply:
Yes, getCloudPlatform will throw an error if a workspace is in the Ready state and has no CloudPlatform.

I agree with your second point, but I didn't know what the UI might do (or not) to prevent the user from trying to use a non-Ready workspace. Would you object to me submitting this as-is and possibly either you or I do a follow-up PR specific to the snapshot service? I would love to get this change in ASAP because it alleviates workspace deletion pain.

Reply:
Sounds good, go for it!